### PR TITLE
docs: Sync Libbeat documentation

### PR DESCRIPTION
**Pull request by bmorelli25**
## Summary

Originally a follow-up to https://github.com/elastic/apm-server/pull/5113, but turned into something more. This PR copies over documentation from the Beats repo.

I _think_ everything in this PR makes sense in the APM Server Reference, but I'd appreciate someone from the apm-server team taking a look.

## More information

This PR backports documentation changes from the following PRs:

* `kafka.asciidoc` 7.12 — https://github.com/elastic/beats/pull/23629 & https://github.com/elastic/beats/pull/12867
* `logging-config.asciidoc` 7.12 — https://github.com/elastic/beats/pull/24207
* `shared-ssl-config.asciidoc` — 7.12 (restructure) https://github.com/elastic/beats/pull/24342
* `shared-path-config.asciidoc` 7.13 — https://github.com/elastic/beats/pull/24334
* `fileout.asciidoc` — master https://github.com/elastic/beats/pull/19347

## Backport to...

Commits in this PR need to be backported to different branches:

- [ ] `7.x`, `7.13`, `7.12` — d77347f & 13b5820
- [ ] `7.x`, `7.13` — a7a7d66 & de79e36
- [x] Do not backport — c7f054f

### CODE REVIEW

File: docs/copied-from-beats/docs/shared-path-config.asciidoc [hunk 1]
```diff
@@ -106,3 +106,20 @@ Example:
...
+This is also used by the system module to read files from `/proc` and `/sys`.
+endif::[]
+
+
+Example:
+
+[source,yaml]
+------------------------------------------------------------------------------
+path.logs: /var/log/beats
+------------------------------------------------------------------------------
```
Comment (axw) [inline docs/copied-from-beats/docs/shared-path-config.asciidoc:125]: this example doesn't match the section

**Review by axw — APPROVED** [review]
Spotted one small thing, which I guess will need fixing in the beats repo. LGTM otherwise

File: docs/copied-from-beats/docs/shared-path-config.asciidoc (hunk 1 above)
Comment (bmorelli25) [reply docs/copied-from-beats/docs/shared-path-config.asciidoc:125]: Ah, good spot 🎯. I'll follow up!

Comment (bmorelli25) [reply docs/copied-from-beats/docs/shared-path-config.asciidoc:125]: Fixed in Beats. Copy/pasting into this PR so we don't have to wait.

File: docs/copied-from-beats/docs/shared-path-config.asciidoc [hunk 2]
```diff
@@ -106,3 +106,20 @@ Example:
...
+ifeval::["{beatname_lc}"=="metricbeat"]
+This is also used by the system module to read files from `/proc` and `/sys`.
+endif::[]
+
+
+Example:
+
+[source,yaml]
+------------------------------------------------------------------------------
+path.logs: /var/log/beats
```
Comment (bmorelli25) [inline docs/copied-from-beats/docs/shared-path-config.asciidoc:124]:
```suggestion
system.hostfs: /mount/rootfs
```

**Pull request merged**